registry.go

```diff
@@ -55,11 +55,15 @@ func NewRegistryConfiguredBy(filePath string) (*Registry, error) {
 }
 
 func NewRegistry(config *Config) *Registry {
+	return NewRegistryWithLogger(config, defaultLogger())
+}
+
+func NewRegistryWithLogger(config *Config, logger Logger) *Registry {
```

Comment:
Would it be cleaner to have the logger function and related information come from the config object? It seems like it is just an aspect of how the registry is configured.

Reply:
Works for me. I'll knock that out shortly ...